qmail-remote.c

```diff
@@ -287,20 +288,20 @@ stralloc canonbox = {0};
 
 void addrmangle(stralloc *saout, char *s)
 {
-  int j;
+  char *at;
  
-  j = str_rchr(s,'@');
-  if (!s[j]) {
+  at = strrchr(s,'@');
+  if (!at) {
```

Comment:
The way you're testing the `str_*` interfaces before changing their implementations makes me very comfortable with those changes. These changes make me nervous without tests around callers. I'm further worried about breaking patches sooner than we have to, particularly for `qmail-remote`. My vote is to save the idea in this commit for later.

Reply:
I have put that commit in it's own branch.